### ISSUE TITLE
No areas of sufficient density - no shapes. Decrease density or area size

### ISSUE THREAD

**Issue by geographika** [hobuinc/hexer#26]
I have built the hexer project on Windows, using GDAL 3.04. The `help` command works fine, however I've not yet been successful in getting any valid outputs.

`curse boundary points.shp --output myboundary.shp` produces an empty shapefile with no records. 

`curse density points.shp --output D:\Temp\myhexagons.shp --edge 100` always ends up returning the following error:

`No areas of sufficient density - no shapes. Decrease density or area size`

I've played around with the `--edge` and `--count` settings, and also run the command on various point shapefiles, and LAS files, including a point shapefile generated with the `make_shapes.py` script. 

Is anyone able to get outputs with recent versions of GDAL, or on Windows? Or maybe I've missed a something basic..

**Comment by abellgithub** [comment]
Does your input shape file consist of points?

**Comment by hobu** [comment]
Something must have rotted. This code hasn't been used in this form for quite a while.

**Comment by hobu** [comment]
This has really rotted. I took a stab at it and couldn't get it working in 15 minutes of futzing. I would try to use PDAL directly to do this nowadays, as that's where this code has migrated to. 

* Write out a text file
* Read the text file with `readers.text`
* Run `pdal boundary` or `pdal tindex` on it.

**Comment by geographika** [comment]
> Does your input shape file consist of points?

Yes, I tried a shapefile generated from a LAS file, a Python generated one, and a couple more point shapefiles I had to hand all with the same result. I also tried converting to web mercator in case that was an issue (and as used at https://gist.github.com/tmcw/7436261). 

> This has really rotted. I took a stab at it and couldn't get it working in 15 minutes of futzing. I would try to use PDAL directly to do this nowadays, as that's where this code has migrated to.
> 
>     * Write out a text file
> 
>     * Read the text file with `readers.text`
> 
>     * Run `pdal boundary` or `pdal tindex` on it.

Thanks @hobu for investigating and the reply. As far as I can tell the `boundary` function has moved to PDAL, but there is no equivalent of the `density` function for creating the hexagon outputs - is this correct?

A shame this project is now deprecated - the `hexer`, `curse` names are inspired!

**Comment by geographika** [comment]
Ignore comment above, the hexagons are available as described at https://pdal.io/en/stable/workshop/exercises/analysis/density/density.html

`pdal density test.las -o  test.shp`

![image](https://user-images.githubusercontent.com/490840/195425349-99b86756-c4d6-45de-adba-5890afbe5799.png)
